### PR TITLE
Military capacity string refactor

### PR DESCRIPTION
**Pull request by hessenfarmer**
first attempt  to fix the regression of #5475

### CODE REVIEW

**Comment by Noordfrees** [comment]
This won't work. For example I quote from the wiki the gaelic plural form rule: 
>    "0 cat", "1 chat", "2 chat", "3 cait", ...

Instead, we'll need to state the raw (unformatted) ngettext strings so we have variables like 
```lua
soldier_text_ctxt = "empire_soldier",
soldier_text_1_untranslated_sg = "%1% soldier (+%2%)",
soldier_text_1_untranslated_pl = "%1% soldiers (+%2%)",
soldier_text_1_translated = npgettext("empire_soldier", "%1% soldier (+%2%)", "%1% soldiers (+%2%)", 0),
soldier_text_2_untranslated_sg = "%1% soldier",
soldier_text_2_untranslated_pl = "%1% soldiers",
soldier_text_2_translated = npgettext("empire_soldier", "%1% soldier", "%1% soldiers", 0),
```
The first two values ensure we have the raw, unlocalized, unformatted strings available in C++. The third, unused value ensures that the .po files contain the full loalized ngettext invocation and not just sg/pl strings. In C++ we can then to
```c++
npgettext(soldier_text_ctxt(), soldier_text_1_untranslated_sg(), soldier_text_1_untranslated_pl(), nr_soldiers)
```
which will use the unlocalized unformatted keys to look up the correct ngettext mapping.

File: src/logic/map_objects/tribes/tribe_descr.cc
Comment (Noordfrees) [inline src/logic/map_objects/tribes/tribe_descr.cc]: We shouldn't require them for add-ons forward compatibility

**Review by Noordfrees — COMMENTED** [review]
:+1:
Not tested yet.

`soldier_singular` and `soldier_plural` can be removed now. The military_capacity scripts can also go.

**Comment by hessenfarmer** [comment]
Not sure whether this works correctly. I tested with the context in npgettext of the military-capacity script, to trigger the usage of the already contained npgettexts in the translations but they did not show up.  
I had not the time to do any further work yesterday on this and probably won't have any time until next week. 
I kept the singular and polural forms as I wanted to build the untranslated strings by them by concatenation in c++. Just for the case that any tribe designer might want to name his soldier different. (e.g. fighter, warrior, etc.).
What is missing as well is the documentation in the amazons units.lua  for these new strings.

feel free to work on this if you have time. otherwise I will look after it next week

**Comment by Noordfrees** [comment]
The reason why it doesn't work is that the existing strings have both different context and different textdomains. I'll look into this now

**Comment by Noordfrees** [comment]
I tested this with custom strings and manually modified PO files, please review

File: data/tribes/initialization/atlanteans/units.lua
Comment (hessenfarmer) [inline data/tribes/initialization/atlanteans/units.lua:199]: Please add some information that the strings must follow exactly the format below and only the name of the soldier can be defined for each tribe. Also add, that the NPGETTEXT entries need to start with the UNUSED_ part

File: data/tribes/initialization/atlanteans/units.lua
Comment (hessenfarmer) [inline data/tribes/initialization/atlanteans/units.lua]: please reword the brackets (the ``npgettext`` results (i.e. the translated pluralized string is not used but by a call in C++)
Please add: the format must be UNUSED_soldier_X = npgettext(soldier_context, soldier_X_sg, soldier_X_pl, 0),
where X is going from 0 to 3 to matchg the c++ call.

**Comment by hessenfarmer** [comment]
code LGTM. 2 small nits for the documentation. 
we can have this for 1.1 but in the longterm we should search for a solution that is not that hard to understand for tribe designers. (i.e. more intuitive)

**Comment by hessenfarmer** [comment]
before merging please remove the military_capacity.lua files
Edit: I can't approve this as I made the PR

**Comment by hessenfarmer** [comment]
PR is fine for me and should be ready. If it is ok I'll merge after the checks are green. I can't formally approve as I made the PR so either @Noordfrees you approve or we merge without formal approval as I would formally approve the work you have done.

**Review by Noordfrees — APPROVED** [review]

**Pull request merged**
